### ISSUE TITLE
is there a possibility to make the panorama as long as the video time?

### ISSUE THREAD

**Issue by installthetech** [installthetech/slit_scan_panorama#4]

**Comment by installthetech** [comment]
Yes, you can do this by changing the "CANVAS" size in DRAWIMAGE.cpp

Try changing the following code:
Mat DRAWIMAGE::DRAW(Mat& thisFrame, int X, int Y, double SCALE, bool SHOWIMAGE = false)
{
  if(CANVAS.empty())
  {
    CANVAS = Mat::zeros(thisFrame.rows, thisFrame.cols*8, CV_8UC3);
  }
...

Change "thisFrame.cols*8" to any suitable number according to your video length. Use higher number to generate longer panorama.

**Comment by zweipunktnull** [comment]
Thank you really much!! This helped me a lot! Thanks for the really fast support!

Cheers

**Comment by zweipunktnull** [comment]
Hey Vinay ;)

Is it possible that you maybe can tell me how your algorithm works? (maybe in 3-4 sentences that I am able to understand?) this would be really nice. Greetings Chris

**Comment by installthetech** [comment]
Hi, Sorry for late reply :P  
Actually, this algorithm works on the fundamental principle of panorama formation i.e. feature matching and homography estimation.
The algorithm goes like this - take two successive image, find features, match features in corresponding images, estimate translation (and rotation) using these features, paste the later image on the new position estimated by homography.

I hope this answers our query, and I think I could help you more precisely if you tell me a bit more about your work.
